js/src/sdk/models/triggers.ts

```diff
@@ -29,18 +40,31 @@ export class Triggers {
    * @returns {CancelablePromise<ListTriggersResponse>} A promise that resolves to the list of all triggers.
    * @throws {ApiError} If the request fails.
    */
-  async list(data: RequiredQuery = {}) {
+  async list(data: TTriggerQuery = {}) {
```

Comment:
The JSDoc comment for the `list` method should be updated to reflect the new parameters and validation requirements. Consider adding:

```typescript
/**
 * Retrieves a list of triggers based on provided filters.
 * @param {TTriggerQuery} data Query parameters for filtering triggers
 * @param {string[]} [data.triggerIds] - Filter by specific trigger IDs
 * @param {string[]} [data.appNames] - Filter by app names
 * @param {string[]} [data.connectedAccountsIds] - Filter by connected account IDs
 * @param {string[]} [data.integrationIds] - Filter by integration IDs
 * @param {boolean} [data.showEnabledOnly] - Show only enabled triggers
 * @throws {CustomError} When no valid filter parameters are provided
 * @returns {Promise<Array>} List of matching triggers
 */
```